### PR TITLE
Onboard postgresql for autogeneration

### PR DESCRIPTION
**Pull request by anthony-c-martin**
```
> azure-schema-generator@1.0.0 generate-single C:\github\azure-resource-manager-schemas\generator
> ts-node cmd/generatesingle "postgresql/resource-manager"

[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git fsck
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git reset -q .
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git checkout -q -- .
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git clean -q -fd
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git gc
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git fetch
[C:\Users\antmarti\AppData\Local\Temp\schm_azspc] executing: git checkout -q origin/master
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\b9jidluxzao --tag=all-api-versions --api-version=2017-12-01-preview --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\postgresql\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[7.68 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\b9jidluxzao\2017-12-01-preview\Microsoft.DBforPostgreSQL.json
Provider Namespace: Microsoft.DBforPostgreSQL
API Version: 2017-12-01-preview
Resource Types (Resource Group Scope):
- servers
- servers/configurations
- servers/databases
- servers/firewallRules
- servers/securityAlertPolicies
- servers/virtualNetworkRules
================================================================================================================================
[C:\github\azure-resource-manager-schemas\generator] executing: autorest-beta.cmd --azureresourceschema --azureresourceschema.multi-scope=true --output-folder=C:\Users\antmarti\AppData\Local\Temp\dhxr73vyg1v --tag=all-api-versions --api-version=2017-12-01 --title=none C:\Users\antmarti\AppData\Local\Temp\schm_azspc\specification\postgresql\resource-manager\readme.md
AutoRest code generation utility [version: 3.0.6116; node: v10.15.3, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      'C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\core\dist' (3.0.6121)
   Loading local AutoRest extension '@autorest/azureresourceschema' (C:\github\azure-resource-manager-schemas\generator\node_modules\@autorest\azureresourceschema)
[8.18 s] Generation Complete
================================================================================================================================
Filename: C:\Users\antmarti\AppData\Local\Temp\dhxr73vyg1v\2017-12-01\Microsoft.DBforPostgreSQL.json
Provider Namespace: Microsoft.DBforPostgreSQL
API Version: 2017-12-01
Resource Types (Resource Group Scope):
- servers
- servers/configurations
- servers/databases
- servers/firewallRules
- servers/securityAlertPolicies
- servers/virtualNetworkRules
================================================================================================================================
```

### CODE REVIEW

**Comment by ifeoluwaokunoren** [comment]
Hi @QingqingYuan2
To simply the schema generation process, ARM is working on an auto-generation pipeline your team can onboard to.
This means once a swagger PR from the [Azure REST-API spec repo](https://github.com/Azure/azure-rest-api-specs/tree/master/specification) is merged either for a new api-version or an update to an existing api-version – the auto generation pipeline would generate the required schemas for that api-version daily – so changes in swagger would be automatically picked up.
The required action is for you to kindly review this PR to ensure that all expected resources have been detected and generated as expected.
You can reach out to me & Anthony over Teams/Email if you have any questions.

File: schemas/2017-12-01-preview/Microsoft.DBforPostgreSQL.json [hunk 1]
```diff
@@ -342,11 +470,15 @@
               "type": "string",
               "enum": [
                 "9.5",
-                "9.6"
+                "9.6",
+                "10",
+                "10.0",
```
Comment (QingqingYuan2) [inline schemas/2017-12-01-preview/Microsoft.DBforPostgreSQL.json]: We'd like to hide these two: 10.0 and 10.2. We'd like to have only 10.

File: schemas/2017-12-01/Microsoft.DBforPostgreSQL.json [hunk 2]
```diff
@@ -532,11 +473,12 @@
                 "9.6",
                 "10",
                 "10.0",
-                "10.2"
+                "10.2",
```
Comment (QingqingYuan2) [inline schemas/2017-12-01/Microsoft.DBforPostgreSQL.json]: Same here. We don't want to expose 10.2

File: schemas/2017-12-01-preview/Microsoft.DBforPostgreSQL.json (hunk 1 above)
Comment (anthony-c-martin) [reply schemas/2017-12-01-preview/Microsoft.DBforPostgreSQL.json]: Swagger is the main API spec for your RP - technically you're already declaring support for 10.2 here. If you need to fix this then I'd suggest submitting a PR to this repo:
https://github.com/Azure/azure-rest-api-specs/tree/master/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL

File: schemas/2017-12-01/Microsoft.DBforPostgreSQL.json (hunk 2 above)
Comment (anthony-c-martin) [reply schemas/2017-12-01/Microsoft.DBforPostgreSQL.json]: (copying previous comment)
Swagger is the main API spec for your RP - technically you're already declaring support for 10.2 here. If you need to fix this then I'd suggest submitting a PR to this repo:
https://github.com/Azure/azure-rest-api-specs/tree/master/specification/postgresql/resource-manager/Microsoft.DBforPostgreSQL

File: schemas/2017-12-01-preview/Microsoft.DBforPostgreSQL.json (hunk 1 above)
Comment (QingqingYuan2) [reply schemas/2017-12-01-preview/Microsoft.DBforPostgreSQL.json]: Thanks! Ife has informed me of this. We're good here.

**Pull request merged**